tb_plugin/README.md

```diff
@@ -41,7 +41,178 @@
+![Alt text](https://github.com/pytorch/kineto/blob/tb_plugin/tb_plugin/docs/images/time_breakdown_priority.PNG)
+
+Performance Recommendation: Leverage the profiling result to automatically give user hints of bottlenecks,  
+and give user feasible suggestions to optimize. 
+ 
+* Operator View
+
+This view displays the performance of every PyTorch operator that is executed either on the host or device.
+
+![Alt text](https://github.com/pytorch/kineto/blob/tb_plugin/tb_plugin/docs/images/operator_view.PNG)
+Each table row is a PyTorch operator, which is a computation operator implemented by C++, 
+such as “aten::relu_”, “aten::convolution”.
+
+Calls: The operator's number of calls.
+
+Device Self Duration: The accumulated time spent on GPU, not including this operator’s child operators.
+
+Device Total Duration: The accumulated time spent on GPU, including this operator’s child operators.
+
+Host Self Duration: The accumulated time spent on Host, not including this operator’s child operators.
+
+Host Total Duration: The accumulated time spent on Host, including this operator’s child operators.
+
+Note: Each above duration means wall-clock time. It doesn't mean the GPU or CPU during this period is fully utilized.
+
+The top 4 pie charts are visualizations of the above 4 columns of durations. 
+They could show each operator's time percentage more straight forward.
+Only N operators with top most durations will be shown in the pie charts. And you can change this N in the text box. 
+
+The search box enables searching operators by name.
+
+“Group By” could choose between “Operator” and “Operator + Input Shape”. 
+The “Input Shape” is shapes of tensors in this operator’s input argument list.
+The empty “[]” means argument with scalar type. 
+For example, “[[32, 256, 14, 14], [1024, 256, 1, 1], [], [], [], [], [], [], []]” 
+means this operator has 9 input arguments, 
+1st is a tensor of size 32\*256\*14\*14, 
+2nd is a tensor of size 1024\*256\*1\*1, 
+the following 7 ones are scalar variables.
+
+![Alt text](https://github.com/pytorch/kineto/blob/tb_plugin/tb_plugin/docs/images/operator_view_group_by_inputshape.PNG)
+
+* Kernel View
+
+This view shows all kernels’ time spent on GPU. The time is got by the kernel's end time minus its start time. 
+
+Note: This view does not include cudaMemcpy or cudaMemset. Because they are not kernels.
+
+![Alt text](https://github.com/pytorch/kineto/blob/tb_plugin/tb_plugin/docs/images/kernel_view.PNG)
+
+Total Duration: The accumulated time of all calls of this kernel.
+
+Mean Duration: The average time duration among all calls. That's "Total Duration" divieded by "Calls".
+
+Max Duration: The maximum time duration among all calls.
+
+Min Duration: The minimum time duration among all calls.
+
+Note: This duration just records a kernel's elapsed time on GPU device. 
+It does not mean the GPU is fully busy during this time interval. 
+In another word, GPU occupancy may be less than 100% during this time interval.
```

Comment:
In strict terminology, when a kernel is running, the GPU utilization shown by nvidia-smi is 100%, but the occupancy may be less than 100%.  "[Occupancy is defined as the ratio of active warps on an SM to the maximum number of active warps supported by the SM](https://docs.nvidia.com/gameworks/content/developertools/desktop/analysis/report/cudaexperiments/kernellevel/achievedoccupancy.htm)". I'll add the link to this word into this doc.

Reply:
Isn't SM Efficiency the metric you're after here? SM Efficiency = active / elapsed cycles, whereas Occupancy = active warps / active cycles / 64. SM Efficiency is a next-level GPU utilization metric (GPU -> SM), whereas occupancy is a bit more hairy - even if you use elapsed cycles as denominator there is not a linear relationship with utilization as you can in theory hit max utilization and throughput with occupancy of 20% or below.

Reply:
Thanks, I get it. Yes, "sm_efficiency" is more suitable here. Occupancy is not a good metric here. But "utilization" also may not be a concise word, in this doc P90, it is defined as "Percent of time over the past second during which one or more kernels was executing on the GPU".
Instead using a terminology word, now I prefer to use naive words to explain it, such as:

> It does not mean the GPU is fully busy on executing instructions during this time interval. 
> Some of the GPU cores may be idle due to such as memory access latency or imbalance workload. 

It will need less expertise knowledge for user to understand. What's your opinion?

Reply:
OK now I understand better what you want to say. Your new suggestion is clearer. My suggestion of SM Efficiency is also not good because it does not capture e.g. stalls due to memory accesses.

If you want to get slightly more technical, how about adding something like this:
"There may be insufficient warps per SMs available for the GPU to effectively hide memory access latencies, or some SMs may be entirely idle due to an insufficient number of blocks. Please refer to Nvidia's best-practices guide: https://docs.nvidia.com/cuda/cuda-c-best-practices-guide/index.html."

Btw, don't remember if you mention it, but the grid and block sizes are useful in estimating theoretical occupancy.

Reply:
Thanks a lot! Added your helpful explanation words.